core/src/main/java/org/apache/brooklyn/policy/AbstractInvokeEffectorPolicy.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.brooklyn.policy;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.Nonnull;
+
+import org.apache.brooklyn.api.effector.Effector;
+import org.apache.brooklyn.api.entity.EntityLocal;
+import org.apache.brooklyn.api.mgmt.Task;
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.core.policy.AbstractPolicy;
+import org.apache.brooklyn.core.sensor.Sensors;
+import org.apache.brooklyn.util.guava.Maybe;
+import org.apache.brooklyn.util.text.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.MoreExecutors;
+
+public abstract class AbstractInvokeEffectorPolicy extends AbstractPolicy {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractInvokeEffectorPolicy.class);
+
+    public static final ConfigKey<AtomicInteger> TASK_COUNTER = ConfigKeys.newConfigKey(AtomicInteger.class,
+            "taskCounter", "Counts the number of incomplete effector invocations made by the policy");
+
+    public static final ConfigKey<String> IS_BUSY_SENSOR_NAME = ConfigKeys.newStringConfigKey(
+            "isBusySensor",
+            "Name of the sensor to publish on the entity that indicates that this policy has incomplete effectors. " +
+                    "If unset running tasks will not be tracked.");
+
+    @Override
+    public void setEntity(EntityLocal entity) {
+        super.setEntity(entity);
+        if (isBusySensorEnabled()) {
+            // Overwrites persisted values and republishes when the entity rebinds.
+            config().set(TASK_COUNTER, new AtomicInteger());
```

Comment:
Should we be storing this as config? On rebind, our effector will not resume so the counter should be reset. However, because we've persisted it as a config key, then it will keep its old value.

Maybe it's better for now to just have this as a field (without `@SetFromFlag` obviously).

What should we do on rebind when we were executing the effector? Currently it will keep the count of 1 and publish busy "true". If we changed to a field, then it would publish "false".